src/service/action-impl.js

```diff
@@ -655,60 +647,90 @@ export class ActionService {
-    const matchActionResults = actionCall.match(actionMacroCall);
-    if (matchActionResults) {
-      const macroActionId = matchActionResults[1];
-      const actionInfo = this.actionMacros_[macroActionId];
-      const ampActionMacroEl = this.ampdoc.getRootNode().querySelector(
-          `amp-action-macro[id=${escapeCssSelectorIdent(macroActionId)}]`);
-      const actionMap = parseActionMap(`action-macro-event:${actionInfo}`,
-          devAssert(ampActionMacroEl));
-      const actionInfoDef = actionMap['action-macro-event'][0];
-      const {target, method} = actionInfoDef;
-      devAssert(actionInfoDef, 'macro action reference does not exist');
-      // Replace the action macro id with the target and method name, as
-      // that is the actual action to be called.
-      const action = actionCall.replace(
-          macroActionId, `${target}.${method}`);
-
-      return {action, actionInfoDef};
-    }
+  parseActionMapForMacro_(event, action, node, opt_args) {
+    const actionMap = parseActionMap(`${event}:${action}`, node);
+    this.setActionMacroArgs_(actionMap[`${event}`],
+        node.hasAttribute('arguments')
+          ? node.getAttribute('arguments').split(',') : [], opt_args);
 
-    return null;
+    return actionMap;
+  }
+
+  /**
+   * Sets the arguments on the action macro actions, taking the arguments
+   * passed by the caller and for the omitted arguments, defaulting to the ones
+   * provided in the macro.
+   * @param {!Array<!Object>} actions
+   * @param {!Array<string>} macroArgs The arguments provided by macro.
+   * @param {JsonObject=} opt_args The arguments provided by the caller.
+   * @private
+   */
+  setActionMacroArgs_(actions, macroArgs, opt_args) {
+    const aliasToArgNameMapping = map();
+    macroArgs.forEach(argVar => {
+      const keyValue = argVar.split('=');
+      const alias = keyValue[0];
+      const argName = keyValue[1];
+      aliasToArgNameMapping[`${alias}`] = argName;
+    });
+    // Iterate over the actions and construct the arguments.
+    actions.forEach(action => {
+      const actionArgs = action.args;
+      for (const arg in actionArgs) {
+        // If the caller has provided the arguments.
+        if (opt_args) {
+          // Then get the actual argument name for the alias used and use this
+          // argument when invoking the action.
+          if (arg in opt_args) {
+            const callerArg = opt_args[`${arg}`];
+            const argName = aliasToArgNameMapping[`${arg}`];
+            actionArgs[`${argName}`] = callerArg;
+            delete actionArgs[`${arg}`];
+          }
+        } else {
+          // Otherwise default to the arguments defined in the macro.
+          const defaultArg = actionArgs[`${arg}`];
+          const argName = aliasToArgNameMapping[`${arg}`];
+          if (argName) {
+            actionArgs[`${argName}`] = defaultArg;
+            delete actionArgs[`${arg}`];
+          }
+        }
+      }
+    });
```

Comment:
I'm not sure you need to handle the method args manually. `tokenizeMethodArguments()` already has some code you can probably use.

For example, we already support parsing [action strings](https://github.com/ampproject/amphtml/blob/master/spec/amp-actions-and-events.md#element-specific-events) like: 
```html
<input on="change: myCarousel.goToSlide(slide=event.value)"></input>
```

Note the `event.value` part is dynamic and evaluated in `dereferenceExprsInArgs()`. So you should be able to use the same mechanism for `amp-action-macro` arguments.

For example:
```html
<amp-action-macro id="myMacro" action="myCarousel.goToSlide(slide=foo)" arguments="foo"></amp-action-macro>

<button on="tap: myMacro(foo=3)"></button>
```

Here, `foo` should be parsed as an `TokenType.ID`. Then, we just need to invoke something like `dereferenceExprsInArgs(args, {foo: 3})` (not exactly this but close). 

Make sense?

Reply:
Yes, gotcha. The problem though is that slide=foo would not evaluate to an expression but would just would evaluate slide as having the string value of 'foo'. We can prepend the argument with 'event' but I don't think that would make sense syntactically. I think 'arg' is a good option and then we can modify the code accordingly to just recognize and evaluate that expression.

    <amp-action-macro id="myMacro" action="myCarousel.goToSlide(slide=arg.foo)" 
    arguments="foo"></amp-action-macro>

    <button on="tap: myMacro(foo=3)"></button>